azerothcore/azerothcore-wotlk
# fix(Core/DB) Align DB Structure - auction house

## Aligning Auction House with Trinity core

##### CHANGES PROPOSED:

 - Stop storing guid for auctioneer.
 - Store instead house ID
 - No separate ID for various houses. Only Horde, Alliance and Neutral.
 - Removed non-needed faction checks.
 - Use enum for auction house IDs

NOTE: This will expire all current auctions and return item to player (or
award to high bidder) in order to prepare database for the changes.


###### ISSUES ADDRESSED:


Closes #1239

##### TESTS PERFORMED:


- Didn't test, just imported

##### HOW TO TEST THE CHANGES:


- Create a auction on both Horde, Alliance and Neutral houses such as Tanaris
- Check the values on the Database for each house
 a. 2 for Alliance
 b. 6 for Horde
 c. 7 for Neutral

##### KNOWN ISSUES AND TODO LIST:


- [x] Going to import file SQL alter table file soon.
- [ ] Missing GetSpawnId in creature.h [Link](https://github.com/TrinityCore/TrinityCore/blob/master/src/server/game/Entities/Creature/Creature.h#L88)
##### Target branch(es):

Master
